### ISSUE TITLE
Type resolve error in Node ESM mode

### ISSUE THREAD

**Issue by otakustay** [yeonjuan/parse-git-diff#29]
For TS project having `"moduleResolution": "NodeNext"` in `tsconfig.json` and `"type": "module"` in `package.json`, TypeScript can't resolve to correct types.

```ts
import parseGitDiffExport from 'parse-git-diff';

// This expression is not callable.
//  Type 'typeof import("node_modules/parse-git-diff/build/types/index")' has no call signatures.ts(2349)
parseGitDiffExport(diff);
```

This is how TypeScript's `NodeNext` mode interacts with CommonJS packages, seems the only fix is to put `index.d.ts` in both `build/cjs` and `build/mjs` and then remove `"types"` key in `exports` field.

**Comment by yeonjuan** [comment]
@otakustay Thank you for reporting the issue. We can reproduce the error. This is clearly a bug and should be fixed!.

**Comment by yeonjuan** [comment]
@otakustay It's fixed and released in v0.0.15! Thanks!
